_source/_posts/2020-10-22-set-up-a-mute-indicator-light-for-zoom-with-hammerspoon.md

```diff
@@ -0,0 +1,283 @@
+
+
+If this doesn't work, it probably means that you need to give Hammerspoon permission to control your computer. To do this, open System Preferences, and go to **Security & Privacy**>**Privacy**" >**Accessibility**, unlock the preference pane to make changes, then select the checkbox next to "Hammerspoon" to allow Hammerspoon to inspect the state of Zoom.
+
+{% img blog/zoom-hammerspoon/console-zoom-inmeeting.png alt:"Image of the Hammerspoon preferences window" width:"512px" em%}{: .center-image }
+
+
+Once you get the `spoon.Zoom:inMeeting()` command working, two others for you to try are `spoon.Zoom:mute()` and `spoon.Zoom:unmute()` try out both while you have a live Zoom meeting running and see what happens!
+
+
+## Configure Hammerspoon to use the Zoom Spoon
+
+At this point, you should be able to run the `spoon.Zoom:mute()` and `spoon.Zoom:unmute()` commands to mute and unmute a Zoom meeting. However, this isn't really any more useful than what you can already do with Zoom, so in this section I'll be showing you how to set up a mute indicator in your menu bar.
+
+**Note:** Zoom provides a global mute and unmute hotkey out of the box! If all you care about is a global mute and unmute hotkey, I suggest using that!
+
+Because I assume that most people don't have a blink() USB LED already, I'm going to show you how to do the next best thing which is to simulate this LED in your menu bar.
+
+To get this set up, use the Hammerspoon menu icon and select "Open Config" to open the `~/.hammerspoon/init.lua` file in your favorite text editor, then edit the file to look like this:
+
+```lua
+
+-- This lets you click on the menu bar item to toggle the mute state
+zoomStatusMenuBarItem = hs.menubar.new(nil)
+zoomStatusMenuBarItem:setClickCallback(function()
+    spoon.Zoom:toggleMute()
+end)
+
+updateZoomStatus = function(event)
+  hs.printf("updateZoomStatus(%s)", event)
+  if (event == "from-running-to-meeting") then
+    zoomStatusMenuBarItem:returnToMenuBar()
+  elseif (event == "muted") then
+    zoomStatusMenuBarItem:setTitle("🔴")
+  elseif (event == "unmuted") then
+    zoomStatusMenuBarItem:setTitle("🟢")
+  elseif (event == "from-meeting-to-running") then
+    zoomStatusMenuBarItem:removeFromMenuBar()
+  end
+end
+
+hs.loadSpoon("Zoom")
+spoon.Zoom:setStatusCallback(updateZoomStatus)
+spoon.Zoom:start()
+
+hs.hotkey.bind('', 'f5', function()
+  spoon.Zoom:toggleMute()
+end)
+```
+
+After you've finished editing the configuration file, save it, then use the Hammerspoon menu icon to select **Reload Config**.
+
+Now it's time to test it out. After selecting "Reload Config" from the Hammerspoon menu icon, reload Zoom and start a meeting. Shortly after the meeting starts, you should see a new menu icon appear in your menu bar. The icon will be green if you are unmuted and red if you are muted. Pressing the `F5` key will toggle between the muted and unmuted states and then update the menu icon as appropriate.
```

Comment:
Buttons and links are typically bolded, not quotes. I'd change "Reload Config" to **Reload Config**, especially since that's how you mention it in the previous paragraph.